### PR TITLE
Stevenic/TimestampProvider class

### PR DESCRIPTION
**Pull request by Stevenic**
Added a new TimestampProvider abstraction to simplify replacing the shared clock.

### CODE REVIEW

**Review by ryanbliss — APPROVED** [review]
LGTM

**Review by huntj88 — APPROVED** [review]
LGTM

File: packages/live-share/src/LiveEvent.ts [hunk 1]
```diff
@@ -236,8 +236,8 @@ export class LiveEvent<
     }
 
     /**
-     * @hidden
-     * Assigns a new timestamp provider.
+     * Assigns a custom timestamp provider.
+     * @provider
```
Comment (corinagum) [inline packages/live-share/src/LiveEvent.ts:240]: Could you run `npm run doctor`?

**Review by corinagum — APPROVED** [review]

File: packages/live-share/src/LiveEvent.ts (hunk 1 above)
Comment (Stevenic) [reply packages/live-share/src/LiveEvent.ts]: done

**Pull request merged**